thrust/examples/sum_columns.cu

```diff
@@ -0,0 +1,79 @@
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/random.h>
+#include <thrust/reduce.h>
+#include <thrust/transform.h>
+#include <thrust/universal_vector.h>
+
+#include <cuda/std/mdspan>
+
+#include <iomanip>
+#include <iostream>
+
+#include "include/host_device.h"
+
+int main()
+{
+  const int rows = 32;
+  const int cols = 16;
+
+  // Create a 2D multidimensional array of ints.
+  thrust::universal_vector<int> data(rows * cols, 42);
+  cuda::std::mdspan M(thrust::raw_pointer_cast(data.data()), rows, cols);
+
+  // Create a range to the flat linear index space for the multidimensional array.
+  auto flat_idx_begin = thrust::make_counting_iterator(0);
+  auto flat_idx_end   = flat_idx_begin + M.size();
+
+  // Fill the array with pseudorandom inputs in parallel on the device.
+  thrust::transform(thrust::device, flat_idx_begin, flat_idx_end, M.data_handle(), [] __host__ __device__(int flat) {
+    thrust::default_random_engine rng;
+    thrust::uniform_int_distribution<int> dist(0, 3);
+    rng.discard(flat); // Advance to the current element's position.
+    return dist(rng);
+  });
```

Comment:
I like the idea of discarding, but could we also seed with the uninitialized data?
```suggestion
  // Fill the array with pseudorandom inputs in parallel on the device.
  thrust::transform(thrust::device, data.begin(), data.end(), data.begin(), [] __host__ __device__(int uninit) {
    thrust::default_random_engine rng{uninit};
    thrust::uniform_int_distribution<int> dist(0, 3);
    return dist(rng);
  });
```

Reply:
Also I do not see the purpose of defining the mdspan already before initializing the data, nothing here is multidimensional

Reply:
We cannot seed with uninitialized data, as that would make the result non-deterministic, which would make it much more difficult to test the example's correctness, as we currently test with FileCheck.